lib/resque/worker.rb

```diff
@@ -235,11 +246,16 @@ def work(interval = 5.0, &block)
       loop do
         break if shutdown?
 
-        unless work_one_job(&block)
+        if paused?
           break if interval.zero?
           log_with_severity :debug, "Sleeping for #{interval} seconds"
-          procline paused? ? "Paused" : "Waiting for #{queues.join(',')}"
+          procline "Paused"
           sleep interval
+        else
+          procline "Waiting for #{queues.join(',')}"
+          unless work_one_job(interval: interval, &block)
+            break if interval.zero?
+          end
```

Comment:
FEATURE: blocking reserve

Reworks `work_one_job` to use the new `blocking_reserve` method. Under the hood, uses `blpop` to efficiently wait for a job in all requested queues.

This is hard to make opt-in, as `blocking_reserve` has different semantics than `reserve`. Is `blpop` universally preferable?